src/platform/web.rs

```diff
@@ -12,7 +12,8 @@ use crate::window::WindowBuilder;
 use web_sys::HtmlCanvasElement;
 
 pub trait WindowExtWebSys {
-    fn canvas(&self) -> HtmlCanvasElement;
+    /// Only returns the canvas if called from inside the window.
+    fn canvas(&self) -> Option<HtmlCanvasElement>;
```

Comment:
This is a breaking change, and should be noted as such in the changelog.

Reply:
Ah, that is probably what you meant there too: https://github.com/rust-windowing/winit/pull/2764#issuecomment-1572739699.

Reply:
Yup

Reply:
Done.